pkg/destroy/aws/aws.go

```diff
@@ -1132,12 +1163,25 @@ func deleteIAM(session *session.Session, arn arn.ARN, logger logrus.FieldLogger)
 	}
 }
 
+func deleteIAMInstanceProfileByName(client *iam.IAM, name *string, logger logrus.FieldLogger) error {
+	_, err := client.DeleteInstanceProfile(&iam.DeleteInstanceProfileInput{
+		InstanceProfileName: name,
+	})
+	if err != nil {
+		if err.(awserr.Error).Code() == iam.ErrCodeNoSuchEntityException {
+			return nil
+		}
+		return err
+	}
+	logger.WithField("name", name).Info("Deleted")
```

Comment:
nit: I'd rather have the caller set this context, like we do [here][1], etc.  The caller can chose which of the information available to it should be logged.  Functions can add additional logging context as new information become available, but they shouldn't write their arguments directly into the logging context.

[1]: https://github.com/openshift/installer/blob/5e34eedac7037178c9d01a0a16cde46d361fc873/pkg/destroy/aws/aws.go#L466